### ISSUE TITLE
`ActionMenu` span showing an axe violation related to ARIA hidden

### ISSUE THREAD

**Issue by jezcommits** [primer/react#3418]
When using `ActionMenu` in a React partial and passing axe scan it is returning an accessibility issue:

> ARIA hidden element must not be focusable or contain focusable elements

This is related to `span.sentinel` and it seems is not related to react partials:

![axe violation related to action menu component](https://github.com/primer/react/assets/8514581/2567a5fb-3302-42d8-a12c-efbffb2998d8)

#### Steps to reproduce

1. Open the new `/issues` experience running `script/setup-hyperlist-web` in codespaces
2. Scan the page with the top-right corner menu open

![axe violation related to action menu in issues page](https://github.com/primer/react/assets/8514581/34251785-83e7-4ce1-aeb4-e1afe9a5d6ca)

Is it possible also scanning [the storybook ](https://primer.style/react/storybook/?path=/story/components-actionmenu--default)
with the menu open

Discussion: https://github.slack.com/archives/C01L618AEP9/p1686844272093569

**Comment by siddharthkp** [comment]
@owenniblock this is a blast from the past! 

> ARIA hidden element must not be focusable or contain focusable elements

Can be reproduced [in storybook](https://primer.style/react/storybook/?path=/story/components-actionmenu-examples--groups-and-descriptions). (see addons → accessibility → incomplete)

These were introduced in https://github.com/primer/behaviors/pull/52 from Feb, 2022

> This change alters the behavior of the focusTrap so that it inserts a sentinel <span> element before and after the trap area. 
> This is because from an Accessibility point of view - trapping focus and not having any focusable elements within that focus is bad for screenreader users.

```html
<span class="sentinel" tabindex="0" aria-hidden="true"></span>
```

However, this fails `aria-hidden-focus` rule from axe-core 4.7:  https://dequeuniversity.com/rules/axe/4.7/aria-hidden-focus

I'm not sure what's the right way to proceed here, is our implementation incorrect or should we ignore this specific rule?

**Comment by owenniblock** [comment]
I would suggest we can ignore this specific rule because when the focus falls on the sentinel, focus is then passed round to the next focusable element within the focusTrap (if memory serves) so the user's focus passes through to the next visible, focusable element.

Having said that, I would love to get feedback from my peers in the Accessibility Team. I'll post in Slack and see if anyone has opinions 🙂

**Comment by owenniblock** [comment]
Interesting, @siddharthkp: one colleague has suggested to try adding role="presentation" to the sentinel elements to see if that makes the axe error go away. If that doesn't work I suggest we can safely ignore this anyway 🙂

**Comment by TylerJDev** [comment]
+1 to what you said @owenniblock.
 
I feel like we utilize this pattern a lot across GitHub. To me it seems like a false positive provided by Axe, that we could probably ignore in cases where 'sentinel' is being used. It would be nice if we could adjust the rule for these cases, but I don't think that this issue comes up enough to justify that now 🤔 .

@jezcommits is this issue a blocker for you? If not, I think we could just ignore the violation.

**Comment by jezcommits** [comment]
Thanks great! thanks for checking. Knowing that this is not affecting the real user, from our side this is not a blocker. We will ignore the violation the next times 👍🏽

**Comment by siddharthkp** [comment]
> I feel like we utilize this pattern a lot across GitHub. To me it seems like a false positive provided by Axe, that we could probably ignore in cases where 'sentinel' is being used. 

Good to know!
